dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/MetadataDumper.java

```diff
@@ -348,6 +308,30 @@ public <T> T runChildTask(Task<T> task) throws MetadataDumperUsageException {
     }
   }
 
+  private File getOutputFile(Connector connector, ConnectorArguments arguments) {
+    // The default output file is based on the connector.
+    // We had a customer request to base it on the database, but that isn't well-defined,
+    // as there may be 0 or N databases in a single file.
+    String defaultFileName = connector.getDefaultFileName(arguments.isAssessment());
+    return arguments
+        .getOutputFile()
+        .map(
+            file -> {
+              String fileName = file.getName();
+              if (StringUtils.endsWithIgnoreCase(fileName, ".zip")) return file;
+              if (file.isFile()) {
+                throw new IllegalStateException(
+                    String.format(
+                        "A file already exists at %1$s. "
+                            + "If you wished to create a directory, please remove/rename the file first. "
```

Comment:
I would suggest to not mix `wished` and `meant`. I think the intention is the same in both cases, so I would stick to only one verb.